src/sql/workbench/browser/modal/modal.ts

```diff
@@ -465,7 +465,7 @@ export abstract class Modal extends Disposable implements IThemable {
 		DOM.append(this.layoutService.container, this._bodyContainer!);
 		this.setInitialFocusedElement();
 
-		this.disposableStore.add(DOM.addDisposableListener(document, DOM.EventType.KEY_UP, (e: KeyboardEvent) => {
+		this.disposableStore.add(DOM.addDisposableListener(document, DOM.EventType.KEY_DOWN, (e: KeyboardEvent) => {
```

Comment:
I just grabbed your branch, cleaned git, built the project and ran it. Using KEY_DOWN, when I tap the Esc key when a callout dialog is open, the cell state moves to inactive -- the edit mode is deactivated. Movie attached to this comment:


https://user-images.githubusercontent.com/12279161/109200900-fae26a80-7755-11eb-9544-0b3e3af4e956.mov

Reply:
@halerankin Let's focus on a fix that can be done at the callout itself that doesn't affect other components (like wizards).

cc @lucyzhang929 @Charles-Gagnon

Reply:
@halerankin Looks like the text cell will go out of editMode when Esc is pressed but pressing Esc a second time does not make the text cell inactive (cell toolbar + blue border is still shown).

Reply:
Okay, so the editor must not go out of edit mode when Esc key is pressed the first time to clear the callout dialog.

Reply:
Is the callout dialog stopping propagation of events when it closes? If it's doing that correctly it should be working correctly.

Although maybe the callout dialog being elsewhere in the DOM is affecting that

Reply:
@lucyzhang929 FYI @halerankin and I looked into this and found out there was an issue with the Notebook key handler so he's going to follow up with a separate fix for that - so you can get your fix in here and then he'll work on the follow up one after.

Reply:
@lucyzhang929 Charles and I looked at the event handling. Your changes are good! 
I will submit a follow-up PR to correct text cell component event handling.

Reply:
@halerankin @Charles-Gagnon is it in addition to https://github.com/microsoft/azuredatastudio/pull/14416?

Reply:
No, separate issue.

Basically there's two problems : 

1. Modal dialog isn't stopping propagation of events
2. It's using KEY_DOWN (with Lucy's revert fix), where as text cell component is using KEY_UP - so even if the modal dialog stops propagation the text cell still gets the other KEY_UP event and handles it separately.

Hale is going to look into having the text cell listen on KEY_DOWN instead + make the modal dialog stop propagation of events. We'll likely want to dig into this a bit more though to make sure we won't regress anything else but I feel like that's the right direction to go.

Reply:
Thanks @Charles-Gagnon. Makes sense.